From 8c61c54ba6248891541d5d849d1d9871ad3c4940 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 07 Sep 2011 15:31:13 +0000
Subject: power: Correctly check for helper exit status

By using WEXITSTATUS() and not relying solely on the
return value of g_spawn_command_line_sync().

https://bugzilla.gnome.org/show_bug.cgi?id=657364
---
diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 80aba9d..af10b4f 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -2254,11 +2254,11 @@ backlight_helper_get_value (const gchar *argument, GError **error)
                                          NULL,
                                          &exit_status,
                                          error);
-        if (!ret)
-                goto out;
-
         g_debug ("executed %s retval: %i", command, exit_status);
 
+        if (!ret || WEXITSTATUS (exit_status) != 0)
+                goto out;
+
         /* parse */
         value = g_ascii_strtoll (stdout_data, &endptr, 10);
 
@@ -2322,10 +2322,12 @@ backlight_helper_set_value (const gchar *argument,
                                          NULL,
                                          &exit_status,
                                          error);
-        if (!ret)
-                goto out;
 
         g_debug ("executed %s retval: %i", command, exit_status);
+
+        if (!ret || WEXITSTATUS (exit_status) != 0)
+                goto out;
+
 out:
         g_free (command);
         return ret;
--
cgit v0.9.0.2
